### ISSUE TITLE
Create `ancestry` endpoint for an Entity

### ISSUE THREAD

**Issue by benjdlambert** [backstage/backstage#7337]
Part of a bigger piece of work, but this has the smallest scope to get us underway with the solution for #7208.

## Feature Suggestion
- Create `/entity/by-name/kind/namespace/name/ancestry` endpoint to show the ancestry.

## Possible Implementation
Still up for debate, but something like this might work nicely.

```ts
interface EntityAncestryResponseItem {
  entity: Entity;
  ancestors: EntityRef[];
}

interface EntityAncestryResponse {
  root: EntityRef;
  items: EntityAncestryResponseItem[];
}
```

**Comment by benjdlambert** [comment]
This would be able to power this #6973 as the `Entity` interface has the `status` on there.

**Comment by kiranpatel11** [comment]
@benjdlambert, as commented in #7208.

Would we be able to use this relationship for "components **_generated/scaffolded-from_** the template".

We are adding a "Components" tab for the template entity, to list the components scaffolded from that particular template. It would help us know the usage/popularity and impact analysis to a template change.

**Comment by freben** [comment]
Closed by #7377

**Comment by freben** [comment]
@kiranpatel11 I believe I saw that that question got answered elsewhere right, in Discord or so? Well for any other interested reader: no, this is not the same thing. This essentially exposes the chain of ancestry in the catalog database, where one entity, while being processed, emits a "child" entity that gets put into the processing queue. This relationship used to be hard to track from the outside.
